task/pylint/0.1/pylint.yaml

```diff
@@ -37,9 +37,16 @@ spec:
   steps:
     - name: lint
       image: docker.io/python:$(inputs.params.PYTHON)
-      workingDir: /workspace/source
+      workingDir: $(workspaces.source.path)
       script: |
         export PATH=$PATH:$HOME/.local/bin
-        pip install -r $(inputs.params.SOURCE_PATH)/$(inputs.params.REQUIREMENTS_FILE)
-        pip show pylint || echo "###\nWarning: Pylint is missing in your requirements\n###" && pip install pylint
+        if [ -n "$(inputs.params.REQUIREMENTS_FILE)" && -e "$(inputs.params.REQUIREMENTS_FILE)" ];then
+           pip install -r $(inputs.params.SOURCE_PATH)/$(inputs.params.REQUIREMENTS_FILE)
+           pip show pylint || {
+              echo "###\nWarning: Pylint is missing in your requirements\n###";
+              pip install pylint
+           }
+        else
+           pip install pylint
```

Comment:
Would it make sense to pin this to a version here?

Reply:
I am not really sure if that's really needed, this is not really common in python world, and if the user really wants to be pinned they can have it in their requirements file..

Reply:
I think we could use this docker image directl too which has the binary https://hub.docker.com/r/cytopia/pylint

but i believe the original author of this task did it like this so the user can choose her preferred python version at task time (which is actually something done quite often in python having the need for different version of python).

Reply:
Yes the idea was here to keep it completely flexible for the user of this task.

As already mentioned by @chmouel we could change the python version for this task (and also the pytest task) to 3.8 or even to latest as it is based on the based python image.